framework/src/modules/token/stores/supported_tokens.ts

```diff
@@ -135,11 +149,17 @@ export class SupportedTokensStore extends BaseStore<SupportedTokensStoreData> {
 		}
 		const [chainID] = splitTokenID(tokenID);
 		let supported: SupportedTokensStoreData = { supportedTokenIDs: [] };
-		try {
-			supported = await this.get(context, chainID);
-		} catch (error) {
-			if (!(error instanceof NotFoundError)) {
-				throw error;
+		const supportExist = await this.has(context, chainID);
+		if (supportExist) {
```

Comment:
i think `has` is not necessary here because below `NotFoundError` is handled and if the data does not exist, it will use the default value above